riscos_toolbox/_types.py

```diff
@@ -0,0 +1,52 @@
+"""RISC OS Toolbox - Types"""
+
+import ctypes
+
+ObjectID = ctypes.c_int32
+ComponentID = ctypes.c_int32
+
+class IDBlock(ctypes.Structure):
+    class Level(ctypes.Structure):
+        _fields_ = [ ("id", ObjectID), ("component", ComponentID) ]
+
+        def __init__(self):
+            self.id = -1
+            self.component = 0
+
+        def __repr__(self):
+            return "{:x}/{}".format(self.id, self.component)
+
+    _fields_ = [ ("ancestor", Level),
+                 ("parent",   Level),
+                 ("self",     Level) ]
+
+    def __init__(self):
+        pass
+
+    def __repr__(self):
+        return "Acestor: {} Parent: {} Self: {}".format(self.ancestor, self.parent, self.self)
+
+class Point(ctypes.Structure):
+    _fields_ = [ ("x", ctypes.c_int), ("y", ctypes.c_int) ]
+
+    def __init__(self, x, y):
+        self.x, self.y = x, y
+
+    def __repr__(self):
+        return "({},{})".format(self.x, self.y)
+
+class BBox(ctypes.Structure):
```

Comment:
In a few places you've used a transform by manually assigning values; you could define the structure here as class Transform or class Matrix to make it easier to use?